var/spack/repos/builtin/packages/py-anuga/package.py

```diff
@@ -13,20 +13,28 @@ class PyAnuga(PythonPackage):
 
     homepage = "https://github.com/GeoscienceAustralia/anuga_core"
     url      = "https://github.com/GeoscienceAustralia/anuga_core/archive/2.1.tar.gz"
+    git      = 'https://github.com/GeoscienceAustralia/anuga_core.git'
 
+    # The git main branch of the repo is now python3-only
+    version('main', branch='main')
     version('2.1', sha256='0e56c4a7d55570d7b2c36fa9b53ee4e7b85f62be0b4c03ad8ab5f51464321d2f')
 
     variant('mpi', default=True, description='Install anuga_parallel')
 
     # At present AnuGA has only been run and tested using python 2.x.
     # We recommend python 2.7.
-    depends_on('python@2.6:2.8', type=('build', 'run'))
-    depends_on('py-setuptools@:44', type='build')
-    depends_on('py-numpy@:1.16', type=('build', 'run'))
+    depends_on('py-setuptools@45:',      when='@2.2:,main', type=('build'))
+    depends_on('py-setuptools@:44.1.0',  when='@:2.1',      type=('build'))
+    depends_on('py-setuptools-scm@:3.5', when='@:2.1',      type=('build'))
+    depends_on('python@3.5:',            when='@2.2:,main', type=('build', 'run'))
+    depends_on('python@2.6:2.7.18',      when='@:2.1',      type=('build', 'run'))
+    depends_on('py-numpy@1.17:',         when='@2.2:,main', type=('build', 'run'))
+    depends_on('py-numpy@:1.16.6',       when='@:2.1',      type=('build', 'run'))
+    depends_on('py-pypar',               when='@:2.1+mpi',  type=('build', 'run'))
+    depends_on('py-future',              when='@2.2:,main', type=('build', 'run'))
     depends_on('py-netcdf4', type=('build', 'run'))
     depends_on('py-matplotlib@:2', type=('build', 'run'))
     depends_on('gdal@:3.2+python', type=('build', 'run'))
-    depends_on('py-pypar', when='+mpi', type=('build', 'run'))
```

Comment:
Still missing a lot of dependencies:
```python
		install_requires=['nose',
                                  'numpy',
                                  'scipy',
                                  'netcdf4',
                                  'matplotlib',
                                  'gdal',
                                  'dill',
                                  'cython',
                                  'future',   # FIXME(Ole): Deprecate
                                  #'openmp',   # FIXME - can't find this. Is it required at this level?
                                  'mpi4py']           
```

Reply:
added, most of them were just for running the tests using the runtests.py, without that added, the build didn't some of them.
```py
+    # Replaces pyar in python3 anuga:
+    depends_on('py-mpi4py',              type=('build', 'run'), when='@2.2:+mpi')
+    depends_on('py-dill',                type=('test'))
+    depends_on('py-nose',                type=('test'))
+    depends_on('py-scipy',               type=('test'))
+    depends_on('py-triangle',            type=('test'))
```
py-triangle is a new package, I add here too.